src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -110,6 +111,7 @@ public static void RecordException(this Activity? activity, Exception? ex)
     /// <remarks> The exception is recorded as per <a href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/exceptions.md">specification</a>.
     /// "exception.stacktrace" is represented using the value of <a href="https://learn.microsoft.com/dotnet/api/system.exception.tostring">Exception.ToString</a>.
     /// </remarks>
+    [Obsolete("Call Activity.AddException instead this method will be removed in a future version.")]
```

Comment:
Should we replace the implementation of this method with a call to `activity.AddException`?

Reply:
It appears that `activity.AddException` is not available in version 8.0.0 and was only introduced starting from version 9.0.0-rc.1.24431.7. Should we consider upgrading our version to `System.Diagnostics.DiagnosticSource v9.0.0-rc.1.24431.7`, or would we prefer to wait for the official release of v9.0.0?

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/5846 will be addressed soon by @rajkumar-rangaraj , so we can wait for that PR to merge first. @ysolomchenko or do you prefer to merge this as-is, and change the implementation to use actiity.AddException as a following after the other changes are in?

Reply:
I think it would be better to wait until #5846 is done before finishing that PR

Reply:
Just merged it 👍